spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1199,3 +1199,108 @@ The following metrics are produced for each channel declared by the application
+
+[source, java]
+----
+
+@Inject
+@Channel("myChannel")
+private Publisher<Message<String>> field;
+
+----
+
+The value _myChannel_ indicates the name of the channel. A channel can be used for the following type:
+
+* Publisher<Message<X>> or Publisher<X>
+* PublisherBuilder<Message<X>> or PublisherBuilder<X>
+* Flowable<Message<X>> or Flowable<X>
+
+
+In addition to the above messaging publishers, an additional `Emitter` is also providied by this specification. For an instance, in order for a method to send messages, you can use _Emitter_, as demonstrated below.
+
+[source, java]
+----
+@Inject @Channel("myChannel")
+Emitter<String> emitter;
+
+public void publishMessage() {
+  emitter.send("a");
+  emitter.send("b");
+  emitter.complete();
+}
+----
+
+You can either inject _Emitter<T>_ or _Emitter<Message<T>>_. 
+When injecting an _Emitter_, you can configure how the backpressure is handled. The _OnOverflow_ annotation configures how the backpressure behaves when emitting messages/payloads faster than the consumption of the messages.
+
+[source, java]
+----
+@Inject @Channel("myChannel")
+@OnOverflow(value=OnOverflow.Strategy.BUFFER, bufferSize=300)
+Emitter<String> emitter;
+
+public void publishMessage() {
+  emitter.send("a");
+  emitter.send("b");
+  emitter.complete();
+}
+----
+
+In the above snippet, the buffer size is set to 300. If `_@OnOverflow_` is absent, the buffer strategy _OnOverflow.Strategy.BUFFER_ will be used. 
+
+If the `bufferSize` is not specified, the size will be the value of the config property _mp.messaging.emitter.defult-buffer-size_.
+If the property does not exist, the default value will be 128 elements. If the buffer is full, an error will be propagated.
+
+The _value_ attribute on `OnOverflow` can be:
+
+* OnOverflow.Strategy.BUFFER - use a buffer, whose size will be determined by the value of _bufferSize_ if set. Otherwise, the size will be the value of the config property _mp.messaging.emitter.defult-buffer-size_ if it exists. Otherwise, 128 will be used. If the buffer is full, an error will be propagated.
+* OnOverflow.Strategy.UNBOUNDED_BUFFER - use an unbounded buffer. If the buffer is full, the applicaiton will die of _OutOfMemory_.
+* OnOverflow.Strategy.DROP - drops the most recent value if the downstream can’t keep up. It means that new value emitted by the emitter are ignored.
+* OnOverflow.Strategy.FAIL - propagates a failure in case the downstream can’t keep up.
+* OnOverflow.Strategy.LATEST - keeps only the latest value, dropping any previous value if the downstream can’t keep up.
+* OnOverflow.Strategy.NONE - ignore the back-pressure signals letting the downstream consumer to implement a strategy.
+
+Here are some examples:
+
+[source, java]
+----
+
+@Inject
+@Channel("myChannel")
+@OnOverflow(value = OnOverflow.Strategy.BUFFER) // Buffer strategy using the buffer size specified by _mp.messaging.emitter.defult-buffer-size_ if exists. Otherwise, 128 will be used.
```

Comment:
default (a is missing)

Reply:
@Emily-Jiang This one has not been fixed.

Reply:
fixed